File: backend/BookstoreAPI/Models/Comprobante.cs
namespace BookstoreAPI.Models
{
    public class Comprobante
    {
        public int Id { get; set; }
        public int Cliente_Id { get; set; }
        public DateTime Fecha { get; set; }
        public string? TipoComprobante { get; set; }
        public string? NumeroComprobante { get; set; }
        public decimal Total { get; set; }
        public string? CAE { get; set; }
        public DateTime? VTO { get; set; }
        public decimal? Bonificacion { get; set; }
        public decimal? PorcentajeBonif { get; set; }
        public decimal? Anticipo { get; set; }
        public decimal? ContraEntrega { get; set; }
        public int? Cuotas { get; set; }
        public decimal? ValorCuota { get; set; }
        public int? Vendedor_Id { get; set; }
        public decimal? GastosEnvio { get; set; }
        public bool EsElectronica { get; set; } = true;
        public bool EsPresupuesto { get; set; } = false;

        // Relación con comprobante asociado (para NC: ID de la factura que cancela)
        public int? ComprobanteAsociado_Id { get; set; }

        // Estado del comprobante: PEN (Pendiente), PAG (Pagada), CAN (Cancelado)
        public string? Estado { get; set; } = "PEN";
    }
}

File: frontend/bookstore-app/src/types/comprobante.ts
export interface Comprobante {
  id: number;
  cliente_Id: number;
  clienteNombre?: string;
  fecha: string;
  tipoComprobante?: string;
  numeroComprobante?: string;
  total: number;
  cae?: string;
  vto?: string;
  bonificacion?: number;
  porcentajeBonif?: number;
  anticipo?: number;
  contraEntrega?: number;
  cuotas?: number;
  valorCuota?: number;
  vendedor_Id?: number;
  vendedorNombre?: string;
  gastosEnvio?: number;
  esElectronica: boolean;
  esPresupuesto: boolean;

  // Estado del comprobante: PEN, PAG, CAN
  estado?: string;

  // Relación con comprobante asociado
  comprobanteAsociado_Id?: number;
  comprobanteAsociadoNumero?: string;  // Para NC: número de la factura

  // Para Facturas: indica si fue cancelada por una NC
  estaCancelado: boolean;
  notaCreditoNumero?: string;  // Número de la NC que canceló esta factura

  detalles: ComprobanteDetalle[];
}

export interface ComprobanteDetalle {
  id?: number;
  articulo_Id: number;
  articuloCodigo?: string;
  articuloDescripcion?: string;
  cantidad: number;
  precio_Unitario: number;
  subtotal: number;
}

export interface CreateComprobanteDto {
  cliente_Id: number;
  fecha: string;
  tipoComprobante?: string;
  numeroComprobante?: string;
  total: number;
  cae?: string;
  vto?: string;
  bonificacion?: number;
  porcentajeBonif?: number;
  anticipo?: number;
  contraEntrega?: number;
  cuotas?: number;
  valorCuota?: number;
  vendedor_Id?: number;
  gastosEnvio?: number;
  esElectronica?: boolean;
  esPresupuesto?: boolean;
  detalles: ComprobanteDetalleDto[];
}

export interface UpdateComprobanteDto {
  cliente_Id: number;
  fecha: string;
  tipoComprobante?: string;
  numeroComprobante?: string;
  total: number;
  cae?: string;
  vto?: string;
  bonificacion?: number;
  porcentajeBonif?: number;
  anticipo?: number;
  contraEntrega?: number;
  cuotas?: number;
  valorCuota?: number;
  vendedor_Id?: number;
  gastosEnvio?: number;
  esElectronica?: boolean;
  esPresupuesto?: boolean;
  detalles: ComprobanteDetalleDto[];
}

export interface ComprobanteDetalleDto {
  articulo_Id: number;
  cantidad: number;
  precio_Unitario: number;
  subtotal: number;
}

export interface ArticuloVendidoZonaItem {
  vendedorInicial: string;
  codigoCliente: string;
  razonSocial: string;
  direccion: string;
  direccionComercial: string;
  descripcionArticulo: string;
  fechaFactura: string;
  numeroFactura: string;
}

export interface ArticulosVendidosZonaReporte {
  zonaId?: number;
  zonaNombre: string;
  items: ArticuloVendidoZonaItem[];
}
